packages/gatsby/src/query/query-watcher.js

```diff
@@ -222,10 +220,6 @@ const watchComponent = componentPath => {
   }
 }
 
-const debounceCompile = _.debounce(() => {
-  updateStateAndRunQueries()
-}, 100)
-
```

Comment:
Is this debounce completely removed or just moved around?

Reply:
We don't need to debounce anymore, because we're not actually compiling on these events, just dispatching an event to the state machine to say that we'd like to compile when ready.